Sources/Editor.php

```diff
@@ -490,188 +791,24 @@ protected function buildButtons(): void
+	 *
+	 * @return void
 	 */
 	protected function buildBbcToolbar(): void
 	{
-		if (!empty(self::$bbc_tags)) {
-			return;
+		if (self::$bbc_tags == []) {
+			self::initBbcTags();
 		}
 
 		Utils::$context['bbc_tags'] = &self::$bbc_tags;
 		Utils::$context['disabled_tags'] = &self::$disabled_tags;
 		Utils::$context['bbc_toolbar'] = &self::$bbc_toolbar;
 		Utils::$context['bbcodes_handlers'] = &self::$bbc_handlers;
 
-		// The below array makes it dead easy to add images to this control. Add it to the array and everything else is done for you!
-		// Note: 'before' and 'after' are deprecated as of SMF 2.1. Instead, use a separate JS file to configure the functionality of your toolbar buttons.
-		/*
-			array(
-				'code' => 'b', // Required
-				'description' => Lang::getTxt('bold', var: 'editortxt'), // Required
-				'image' => 'bold', // Optional
-				'before' => '[b]', // Deprecated
-				'after' => '[/b]', // Deprecated
-			),
-		*/
-		self::$bbc_tags[] = [
-			[
-				'code' => 'bold',
-				'description' => Lang::getTxt('bold', var: 'editortxt'),
-			],
-			[
-				'code' => 'italic',
-				'description' => Lang::getTxt('italic', var: 'editortxt'),
-			],
-			[
-				'code' => 'underline',
-				'description' => Lang::getTxt('underline', var: 'editortxt'),
-			],
-			[
-				'code' => 'strike',
-				'description' => Lang::getTxt('strikethrough', var: 'editortxt'),
-			],
-			[
-				'code' => 'superscript',
-				'description' => Lang::getTxt('superscript', var: 'editortxt'),
-			],
-			[
-				'code' => 'subscript',
-				'description' => Lang::getTxt('subscript', var: 'editortxt'),
-			],
-			[
-				'image' => 'tt',
-				'code' => 'tt',
-				'description' => Lang::getTxt('tt', var: 'editortxt'),
-			],
-			[
-				'image' => 'hidden',
-				'code' => 'spoiler',
-				'description' => Lang::getTxt('spoiler', var: 'editortxt'),
-			],
-			[],
-			[
-				'code' => 'pre',
-				'description' => Lang::getTxt('preformatted_text', var: 'editortxt'),
-			],
-			[
-				'code' => 'left',
-				'description' => Lang::getTxt('align_left', var: 'editortxt'),
-			],
-			[
-				'code' => 'center',
-				'description' => Lang::getTxt('center', var: 'editortxt'),
-			],
-			[
-				'code' => 'right',
-				'description' => Lang::getTxt('align_right', var: 'editortxt'),
-			],
-			[
-				'code' => 'justify',
-				'description' => Lang::getTxt('justify', var: 'editortxt'),
-			],
-			[],
-			[
-				'code' => 'font',
-				'description' => Lang::getTxt('font_name', var: 'editortxt'),
-			],
-			[
-				'code' => 'size',
-				'description' => Lang::getTxt('font_size', var: 'editortxt'),
-			],
-			[
-				'code' => 'color',
-				'description' => Lang::getTxt('font_color', var: 'editortxt'),
-			],
-		];
-
-		if (empty(Config::$modSettings['disable_wysiwyg'])) {
-			self::$bbc_tags[\count(self::$bbc_tags) - 1][] = [
-				'code' => 'removeformat',
-				'description' => Lang::getTxt('remove_formatting', var: 'editortxt'),
-			];
-		}
-
```

Comment:
Is there any intention to add back the remove formatting option which appears to be used when the editor is disabled?